mne_connectivity/spectral/tests/test_spectral.py

```diff
@@ -1171,3 +1278,10 @@ def test_save(tmp_path):
         epochs, fmin=(4, 8, 13, 30), fmax=(8, 13, 30, 45),
         faverage=True)
     conn.save(tmp_path / 'foo.nc')
+
+    # multivariate connectivity
+    # use ragged indices & MIC to test padding of indices and patterns
+    indices = (np.array([[0, 1]]), np.array([[2]]))
+    conn_mvc = spectral_connectivity_epochs(
+        epochs, method="mic", indices=indices, sfreq=sfreq, fmin=10, fmax=40)
+    conn_mvc.save(tmp_path / 'foo_mvc.nc')
```

Comment:
Should we move this to a separate test and also test roundtrip IO? I.e. reading in the file after saving results in everything being the same as what you would expect?

Reply:
Yes, good idea. I have added a new test for saving the multivariate results and checking their integrity:
https://github.com/mne-tools/mne-connectivity/blob/409c2c664d64ebc2ff1b36b9e935aab7972fc24c/mne_connectivity/spectral/tests/test_spectral.py#L1286-L1315

Reply:
When adding this test, I realised that `read_connectivity` returns an object whose `indices` attribute is an array, whereas it is a tuple of arrays/lists in the original object.

Because `indices` is an array, a later check in `_check_data_consistency` that `indices` is a tuple fails (L538), and so it tries to check if the array is a string (L553), which leads to an elementwise comparison that fails. With my NumPy version (1.24.3) I get the following warning which causes pytest to fail: `FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison`, so this should cause a straight-up error in newer NumPy versions.

https://github.com/mne-tools/mne-connectivity/blob/409c2c664d64ebc2ff1b36b9e935aab7972fc24c/mne_connectivity/base.py#L538-L561

The same issue also occurs for the pre-existing bivariate connectivity measures, however since reading saved connectivity objects is only tested where `indices=None`, this warning is never seen.

A solution is to convert `indices` from an array to a tuple of arrays/lists when reading connectivity objects in `_xarray_to_conn` (i.e. return it to the same format as it was in the original connectivity object).
https://github.com/mne-tools/mne-connectivity/blob/409c2c664d64ebc2ff1b36b9e935aab7972fc24c/mne_connectivity/io.py#L56-L58